tests/remotes/hdfs.py

```diff
@@ -115,10 +120,19 @@ def _get(url, tar, target):
     os.environ["HADOOP_HOME"] = hadoop_home
     os.environ["PATH"] += f":{hadoop_home}/bin:{hadoop_home}/sbin"
 
+    # NOTE: must set CLASSPATH to connect using pyarrow.fs.HadoopFileSystem
+    result = subprocess.run(
+        [f"{hadoop_home}/bin/hdfs", "classpath", "--glob"],
+        universal_newlines=True,
+        stdout=subprocess.PIPE,
+        check=False,
+    )
+    os.environ["CLASSPATH"] = result.stdout
```

Comment:
@efiop, thanks for reviewing my PR. This can be removed. I realized after I opening my PR that those lines of code were probably just "fixing" my configuration problems. Can I make a commit removing those lines?

Reply:
@jwokaty Sure!

Reply:
@efiop After removing these lines, there are [timeout errors](https://github.com/iterative/dvc/pull/4973/checks?check_run_id=1486855831#step:7:23337). It also complains that CLASSPATH isn't set:

```
Environment variable CLASSPATH not set!
getJNIEnv: getGlobalJNIEnv failed
/arrow/cpp/src/arrow/filesystem/hdfs.cc:51: Failed to disconnect hdfs client: IOError: HDFS hdfsFS::Disconnect failed, errno: 255 (Unknown error 255) Please check that you are connecting to the correct HDFS RPC port
```

The [new interface needs CLASSPATH set](https://arrow.apache.org/docs/python/filesystems.html#hadoop-file-system-hdfs). It also states that it should be set automatically (how?), but that doesn't happen. I've tried setting it the docker by adding an env_file to docker-compose.yml and can inspect that it is set in the container; however, that didn't satisfy the requirement to connect to HDFS. Could you please advise?

Reply:
@jwokaty Yeah, odd. Previous docs also say that about `connection`, but for some reason, it works fine there. You might be onto pyarrow bug. I would trace the logic with old `connection` and new `HadoopFileSystem` and see where the difference is.